internal/receiver/smartagentreceiver/event.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package smartagentreceiver
+
+import (
+	"fmt"
+
+	"github.com/signalfx/golib/v3/event"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+const (
+	SFxEventCategoryKey   = "com.splunk.signalfx.event_category"
+	SFxEventPropertiesKey = "com.splunk.signalfx.event_properties"
+)
+
+// eventToLog converts a SFx event to a pdata.Logs entry suitable for consumption by LogConsumer.
+// based on https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5de076e9773bdb7617b544a57fa0a4b848cec92c/receiver/signalfxreceiver/signalfxv2_event_to_logdata.go#L27
```

Comment:
Do you know if it would be possible to just reuse the method from the signalfxreceiver package? I know it would need to be exposed but is that something we should consider given that the conversion logic would essentially be the same?

Reply:
nvm, read the PR description :)

Reply:
I considered updating the agent to use the protobuf type but this code would then be in that project and I'd have to export the receiver helper.  Seemed simpler to do it here.